packages/components/src/notice/types.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import type { MouseEventHandler, ReactNode } from 'react';
+
+type CommonNoticeActionProps = {
+	label: string;
+	className?: string;
+	noDefaultClasses?: boolean;
+	variant?: 'primary' | 'secondary' | 'link';
+};
+// `url` and `onClick` can both be provided, but `url` takes precedence. If
+// `url` is provided, the action's button will be rendered as an anchor and
+// `onClick` will be ignored.
+type NoticeActionWithURL = CommonNoticeActionProps & {
+	url: string;
+	onClick?: never;
+};
+type NoticeActionWithOnClick = CommonNoticeActionProps & {
+	url?: never;
+	onClick: MouseEventHandler< HTMLButtonElement >;
+};
+
+export type NoticeAction = NoticeActionWithURL | NoticeActionWithOnClick;
+
+export type NoticeChildren = string | JSX.Element;
+
+export type NoticeProps = {
+	/**
+	 * A CSS `class` to give to the wrapper element.
+	 */
+	className?: string;
+	/**
+	 * The displayed message of a notice. Also used as the spoken message for
+	 * assistive technology, unless `spokenMessage` is provided as an alternative message.
+	 */
+	children: ReactNode;
+	/**
+	 * Used to provide a custom spoken message in place of the `children` default.
+	 *
+	 * @default `children`
+	 */
+	spokenMessage?: ReactNode;
+	/**
+	 * Can be warning (yellow), success (green), error (red), or info.
+	 *
+	 * @default 'info'
+	 */
+	status?: 'warning' | 'success' | 'error' | 'info';
+	/**
+	 * Function called when dismissing the notice
+	 *
+	 * @default noop
+	 */
+	onRemove?: () => void;
+	/**
+	 * A politeness level for the notice's spoken message. Should be provided as
+	 * one of the valid options for an aria-live attribute value. If not provided,
+	 * a sensible default is used based on the notice status. Note that this
+	 * value should be considered a suggestion; assistive technologies may
+	 * override it based on internal heuristics.
+	 *
+	 * @ default - 'assertive' for 'error' status
+	 * 			 - 'polite' for all other statuses
```

Comment:
The space in between the `@` and the `default` doesn't allow TypeScript / JSDoc to pick that up correctly

<img width="611" alt="Screenshot 2023-01-27 at 17 21 51" src="https://user-images.githubusercontent.com/1083581/215137949-c1bd9a14-8c36-42c8-a2e4-c78039416280.png">

We can adjust the spacing (and keep it all on one line)

```suggestion
	 * @default 'assertive' for 'error' status, 'polite' for all other statuses
```

<img width="598" alt="Screenshot 2023-01-27 at 17 22 14" src="https://user-images.githubusercontent.com/1083581/215138084-8a2977f9-8509-4039-bca8-2ebf32ac420f.png">

This doesn't seem to make any differences on Storybook, which is more based on the actual code (and in fact shows `getDefaultPoliteness( status )` as the default value for the prop)

<img width="813" alt="Screenshot 2023-01-27 at 17 17 17" src="https://user-images.githubusercontent.com/1083581/215138543-9c590add-94c0-4b99-8411-6bddfc78f46b.png">

Reply:
oops, that space was a typo, my bad. I've fixed it and one-lined the default as you suggested.